packages/features/eventtypes/components/MultiplePrivateLinksController.tsx

```diff
@@ -1,97 +1,348 @@
+                    : t("expires_on_date", { date: expiryDate });
+                } else if (
+                  val.maxUsageCount !== undefined &&
+                  val.maxUsageCount !== null &&
+                  !isNaN(Number(val.maxUsageCount))
+                ) {
+                  // Calculate uses and determine if expired
+                  const maxUses = val.maxUsageCount;
+                  const usedCount = latestUsageCount;
+                  const remainingUses = maxUses - usedCount;
+
+                  // Link is expired if usage count has reached or exceeded the limit
+                  isExpired = usedCount >= maxUses;
+
+                  if (isExpired) {
+                    linkDescription = t("usage_limit_reached");
+                  } else {
+                    linkDescription = t("remainder_of_maximum_uses_left", {
+                      remainder: remainingUses,
+                      maximum_uses: `${maxUses} ${remainingUses === 1 ? "use" : "uses"}`,
+                    });
+                  }
+                }
+
                 return (
-                  <li data-testid="add-single-use-link" className="mb-4 flex items-center" key={val}>
-                    <TextField
-                      containerClassName="w-full"
-                      disabled
-                      data-testid={`generated-hash-url-${key}`}
-                      labelSrOnly
-                      type="text"
-                      defaultValue={singleUseURL}
-                      addOnSuffix={
-                        <Tooltip content={t("copy_to_clipboard")}>
+                  <li data-testid="add-single-use-link" className="mb-4 flex flex-col" key={val.link}>
+                    <div className="flex items-center">
+                      <TextField
+                        containerClassName={classNames("w-full")}
+                        disabled={isExpired}
+                        value={singleUseURL}
+                        readOnly
+                        className={classNames(isExpired ? "bg-red-50 text-gray-400" : "bg-gray-50")}
+                        data-testid="private-link-url"
+                        addOnSuffix={
+                          !isExpired ? (
+                            <Tooltip content={t("copy_link")}>
+                              <Button
+                                type="button"
+                                color="minimal"
+                                size="sm"
+                                StartIcon="copy"
+                                onClick={() => {
+                                  navigator.clipboard.writeText(singleUseURL);
+                                  showToast(t("link_copied"), "success");
+                                }}
+                              />
+                            </Tooltip>
+                          ) : (
+                            <Badge variant="red">{t("expired")}</Badge>
```

Comment:
Let's sort all the expired link either at the bottom or top so there aren't mixed in between active links
![Screenshot 2025-07-08 at 1 42 38 PM](https://github.com/user-attachments/assets/726fe9df-686d-4da8-8f6f-b19596a32f19)